File: math/primenumbers/136-single-number/single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    
    let map={}

    for(let i=0;i<nums.length;i++){
        if(map[nums[i]]){
            map[nums[i]]++
        }
        else{
            map[nums[i]]=1
        }
    }
    console.log(map)

    for(let m in map){
        if(map[m]==1){
            return m
        }
    }
};

File: math/primenumbers/136-single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {

        HashMap<Integer,Integer> map=new HashMap<>();

        for(int i=0;i<nums.length;i++){
            if(map.containsKey(nums[i])){
                map.put(nums[i],map.get(nums[i])+1);
            }
            else{
                map.put(nums[i],1);
            }
        }

        
        for(Map.Entry<Integer,Integer> pair:map.entrySet()){
           if(pair.getValue()==1){
             return pair.getKey();
           }
        }


        return -1;
        
    }
}